Podcast/SeriesDetailViewController.swift

```diff
@@ -76,51 +78,47 @@ class SeriesDetailViewController: ViewController, SeriesDetailHeaderViewDelegate
         
         loadingAnimation.startAnimating()
         
+        if let series = series {
+            setSeries(series: series)
+        }
     }
     
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewDidAppear(animated)
-        
-        guard let series = self.series else { return }
-        
-        self.loadingAnimation.stopAnimating()
-        self.seriesHeaderView.isHidden = false
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
         
+        guard let series = series else { return }
+        updateSeriesHeader(series: series)
         episodeTableView.reloadData()
-        
-        // check before reloading data whether the Player has stopped playing the currentlyPlayingIndexPath
-        if let indexPath = currentlyPlayingIndexPath {
-            let episode = series.episodes[indexPath.row]
-            if Player.sharedInstance.currentEpisode?.id != episode.id {
-                currentlyPlayingIndexPath = nil
-            }
-        }
     }
     
     // use if creating this view from just a seriesID
-    func fetchAndSetSeries(seriesID: String) {
+    func fetchSeries(seriesID: String) {
         let seriesBySeriesIdEndpointRequest = FetchSeriesForSeriesIDEndpointRequest(seriesID: seriesID)
 
         seriesBySeriesIdEndpointRequest.success = { (endpointRequest: EndpointRequest) in
             guard let series = endpointRequest.processedResponseValue as? Series else { return }
-            self.setSeries(series: series)
+            self.updateSeriesHeader(series: series)
+            self.loadingAnimation.stopAnimating()
+            print("Stop Loading")
         }
         
         System.endpointRequestQueue.addOperation(seriesBySeriesIdEndpointRequest)
+        
+        fetchEpisodes()
     }
     
-    func setSeries(series: Series) {
+    func updateSeriesHeader(series: Series) {
         self.series = series
         title = series.title
+        seriesHeaderView.setSeries(series: series)
+        self.seriesHeaderView.isHidden = false
+        self.seriesHeaderView.sizeToFit()
```

Comment:
if seriesHeaderView has snapkit constraints we shouldn't have to call sizeToFit()

Reply:
i don't think you need self here for seriesheaderview

Reply:
Lol I'm not sure I was copying and pasting from an old function. Will change.